Development dashboard

ensist opened the pull request:
...

jonathan016 committed:
Added entity models for dashboard API

Entity models for dashboard API includes AssetModel and RequestModel, as
status field in dashboard contains information about asset and requests.

jonathan016 committed:
Added repository classes for dashboard page

Repository classes for asset and request have been created, with
addition of some methods.

jonathan016 committed:
Added files for dashboard feature

Dashboard feature files have been added and the output for status field
is done. The API, however, has slight changes, but development team has
adjusted to the API.

jonathan016 committed:
Added early dashboard feature

Early dashboard feature is added here, but more changes are expected,
such as faster searching process, etc. Several other features are also
slightly changed, with most changes being renaming.

jonathan016 committed:
Changed code data type and renamed variables

Code variable, located at BaseResponse is changed from having data type
String to Integer. Variables in LoginController is also renamed for
increased readability.

jonathan016 committed:
Finished dashboard API & edited login API

Dashboard API has been defined, but needs further review. Login API is
also slightly changed, with the change being several code refactoring.

ensist commented on model/pom.xml:
```diff
@@ -5,7 +5,7 @@
         <groupId>com.oasis</groupId>
         <artifactId>oasis</artifactId>
-        <version>1.0.0-SNAPSHOT</version>
+        <version>1.1.0-SNAPSHOT</version>
```
no need to change the version, you can still use the old version

ensist commented on model/src/main/java/com/oasis/model/FieldName.java:
```diff
@@ -11,4 +11,24 @@
     public static final String EMPLOYEE_SUPERVISION_ID = "supervisionId";
+
+    //Field names for AssetModel
+    public static final String ASSET_NAME = "name";
```
separate it to another classes
AssetFieldName, RequestFieldName, etc

jonathan016 replied:
Commit ID: 57faedf

ensist commented on model/src/main/java/com/oasis/model/entity/AssetModel.java:
```diff
@@ -0,0 +1,30 @@
+    private String location;
+
+    @Field(FieldName.ASSET_PRICE)
+    private Double price;
```
price should not be null, use primitive data type double

jonathan016 replied:
Commit ID: 5a96711

ensist commented on model/src/main/java/com/oasis/model/entity/AssetModel.java:
```diff
@@ -0,0 +1,30 @@
+    private Double price;
+
+    @Field(FieldName.ASSET_STOCK)
+    private Integer stock;
```
use primitive int

jonathan016 replied:
Commit ID: 5a96711

ensist commented on pom.xml:
```diff
@@ -23,13 +23,18 @@
         <jackson.version>2.9.4</jackson.version>
         <lombok.version>1.18.2</lombok.version>
+        <orika.version>1.5.2</orika.version>
+        <model.version>${project.parent.version}</model.version>
```
no need to separate model.version, repository.version, etc.
all of those will always have same version

jonathan016 replied:
Commit ID: a81d3e2

ensist commented on repository/src/main/java/com/oasis/repository/AssetRepository.java:
```diff
@@ -0,0 +1,13 @@
+
+@Repository
+public interface AssetRepository extends MongoRepository<AssetModel, String> {
+    List<AssetModel> findAllByStockGreaterThan(Integer stockLimit);
```
add a new line here

ensist replied:
use primitive int

jonathan016 replied:
Commit ID: 5a96711

jonathan016 replied:
Where should the new line be Kak? After line 10 or 11?

ensist commented on repository/src/main/java/com/oasis/repository/AssetRepository.java:
```diff
@@ -0,0 +1,13 @@
+@Repository
+public interface AssetRepository extends MongoRepository<AssetModel, String> {
+    List<AssetModel> findAllByStockGreaterThan(Integer stockLimit);
+    AssetModel findBy_id(String assetId);
```
I think it should be findById

jonathan016 replied:
This is fixed in assets feature

ensist commented on repository/src/main/java/com/oasis/repository/RequestRepository.java:
```diff
@@ -0,0 +1,12 @@
+
+@Repository
+public interface RequestRepository extends MongoRepository<RequestModel, String> {
+    List<RequestModel> findAllByStatusAndEmployeeId(String status, String employeeId);
```
switch the parameter, employeeId first then status.
prefer to read it from the most unique one
findAllByEmployeeIdAndStatus

jonathan016 replied:
Commit ID: a81d3e2

ensist commented on service/src/main/java/com/oasis/service/RoleDeterminer.java:
```diff
@@ -0,0 +1,32 @@
+            for (SupervisionModel supervision : supervisions) {
+                EmployeeModel supervisedEmployee = employeeRepository.findBy_id(supervision.getEmployeeId());
+                if (supervisedEmployee.getSupervisingCount() > 0) {
+                    return ROLE_ADMINISTRATOR;
```
wrong logic for administrator, just add a new field isAdmin in employee collection or create a new table to store admin username and password

ensist commented on service/src/main/java/com/oasis/service/ServiceConstant.java:
```diff
@@ -0,0 +1,11 @@
+package com.oasis.service;
+
+public class ServiceConstant {
+    public static final Integer STOCK_LIMIT = 0;
```
Rename to `ZERO`

jonathan016 replied:
Commit ID: 5a96711

ensist commented on service/src/main/java/com/oasis/service/api/DashboardServiceApi.java:
```diff
@@ -0,0 +1,20 @@
+import java.util.List;
+
+public interface DashboardServiceApi {
+    Integer getAvailableAssetsCount();
```
add a new line below method declaration

jonathan016 replied:
Commit ID: a81d3e2

ensist commented on web-controller/src/main/java/com/oasis/constant/ControllerConstant.java:
```diff
@@ -0,0 +1,5 @@
+package com.oasis.constant;
+
+public class ControllerConstant {
+    public static final Integer DASHBOARD_REQUEST_UPDATE_PAGE_SIZE = 5;
```
change to primitive int

jonathan016 replied:
Commit ID: 5a96711

ensist commented on web-controller/src/main/java/com/oasis/constant/ErrorCodeAndMessage.java:
```diff
@@ -7,4 +7,10 @@
     public static final String[] PASSWORD_DOES_NOT_MATCH = {
```
change from array of string to object
ErrorCode {
String code;
String message;
}

jonathan016 replied:
Commit ID: 4a81232 (if not mistaken)

ensist commented on web-controller/src/main/java/com/oasis/webcontroller/DashboardController.java:
```diff
@@ -0,0 +1,285 @@
+    private DashboardServiceImpl dashboardServiceImpl;
+
+    @GetMapping(value = APIMappingValue.API_DASHBOARD_STATUS,
+            produces = APPLICATION_JSON_VALUE)
```
add consumes

jonathan016 replied:
Commit ID: 4a81232

ensist commented on web-controller/src/main/java/com/oasis/webcontroller/DashboardController.java:
```diff
@@ -0,0 +1,285 @@
+        );
+    }
+
+    private NoPagingResponse<DashboardStatusResponse>
```
separate it to different class

jonathan016 replied:
Commit ID: 4a81232

ensist commented on web-controller/src/main/java/com/oasis/webcontroller/DashboardController.java:
```diff
@@ -0,0 +1,285 @@
+        return successResponse;
+    }
+
+    private PagingResponse<DashboardRequestUpdateResponse>
```
separate it to different class

jonathan016 replied:
Commit ID: 4a81232

ensist commented on web-controller/src/main/java/com/oasis/webcontroller/DashboardController.java:
```diff
@@ -0,0 +1,285 @@
+        return successResponse;
+    }
+
+    private PagingResponse<FailedResponse>
```
separate it to different class

jonathan016 replied:
Commit ID: 4a81232

ensist commented on web-controller/src/main/java/com/oasis/webcontroller/DashboardController.java:
```diff
@@ -0,0 +1,285 @@
+        return failedResponse;
+    }
+
+    private void fillData(List<RequestModel> requestUpdates, String employeeId, String currentTab, String role) {
```
separate it to different class

jonathan016 replied:
Commit ID: 4a81232

jonathan016 committed:
Moved field name variables

Field name constant variables have been moved to separate classes.

jonathan016 committed:
Changed data type & deleted unused files

Wrapper classes used as data type are changed to primitive data type to
avoid null value. Unused files are also deleted, namely response for
dashboard status and request update section.

jonathan016 committed:
Managed modules, renamed method, add new lines

Modules versions have been changed to follow project.parent.version
value. A method is also renamed to show uniqueness
(findAllByEmployeeIdAndStatus). New lines for DashboardServiceApi is also
added.

jonathan016 committed:
Fixed some issues from pull request

Several issues stated by mentor in the last pull request (#5) are fixed.

jonathan016 committed:
Refactored code for dashboard feature

Codes which were in DashboardController class is moved to
DashboardServiceImpl as the codes are related to logic of the
application.

jonathan016 committed:
Reformatted code for dashboard feature

Dashboard feature's codes are reformatted for increased readability.
